arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -217,51 +261,31 @@ class ContentTableViewController: UITableViewController, CustomSearchHeaderViewD
             tableView.reloadRows(at: [previouslyExpandedIndexPath, indexPath], with: .fade)
         }
     }
-    
-    //MARK: - CustomSearchHeaderViewDelegate
-    
-    func customSearchHeaderViewWillShowSuggestions(_ customSearchHeaderView: CustomSearchHeaderView) {
-        var headerViewFrame = self.headerView.frame
-        headerViewFrame.size.height = customSearchHeaderView.expandedViewHeight
-        
-        UIView.animate(withDuration: 0.3, delay: 0, options: [], animations: { () -> Void in
-            self.headerView.frame = headerViewFrame
-            self.tableView.tableHeaderView = self.headerView
-        }, completion: nil)
-    }
-    
-    func customSearchHeaderViewWillHideSuggestions(_ customSearchHeaderView: CustomSearchHeaderView) {
-        var headerViewFrame = self.headerView.frame
-        headerViewFrame.size.height = customSearchHeaderView.shrinkedViewHeight
 
-        UIView.animate(withDuration: 0.3, delay: 0, options: [], animations: { () -> Void in
-            self.headerView.frame = headerViewFrame
-            self.tableView.tableHeaderView = self.headerView
-        }, completion: nil)
+}
+
+//MARK: - DownloadProgressViewDelegate
+extension ContentTableViewController: DownloadProgressViewDelegate {
+
+    func downloadProgressViewDidCancel(downloadProgressView: DownloadProgressView) {
+        self.bundleResourceRequest.progress.cancel()
+        self.bundleResourceRequest?.endAccessingResources()
     }
+
+}
+
+extension ContentTableViewController: UISearchResultsUpdating {
     
-    func customSearchHeaderView(_ customSearchHeaderView: CustomSearchHeaderView, didFindSamples sampleNames: [String]?) {
-        if let sampleNames = sampleNames {
-            let samples = samplesByNames(sampleNames)
-            if !samples.isEmpty {
-                //show the results
-                let controller = self.storyboard!.instantiateViewController(withIdentifier: "ContentTableViewController") as! ContentTableViewController
-                controller.samples = samples
-                controller.title = "Search results"
-                controller.containsSearchResults = true
-                self.navigationController?.show(controller, sender: self)
-                return
-            }
+    func updateSearchResults(for searchController: UISearchController) {
+        if searchController.isActive,
+            let query = searchController.searchBar.text?.trimmingCharacters(in: .whitespacesAndNewlines),
+            !query.isEmpty{
+            samples = searchEngine!.sortedSamples(matching: query)
         }
-        
-        SVProgressHUD.showError(withStatus: "No match found")
-        
+        else{
+            samples = searchEngine!.samples
```

Comment:
I would strongly suggestion using a separate property for the found samples rather than blowing away the master list. Yes, you passed it onto `SearchEngine`, but I would argue that that property should be private.